src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -4,70 +4,198 @@
 # --------------------------------------------------------------------------------------------
 
 from ._util import ParametersContext, patch_arg_make_required
+from azure.cli.core.commands import CliArgumentType
 
+#####
+#           Reusable param type definitions
+#####
+
+server_param_type = CliArgumentType(options_list=('--server', '-s'), help='Name of the Azure SQL server.')
+
+###############################################
+#                sql db                       #
+###############################################
+
+# sql db params that are used for non-default create modes. These are ignored
+# for regular create (because we have custom commands for special create modes)
+# and ignored for update (because they are not applicable to update)
+sql_db_special_create_mode_params = ['create_mode', 'source_database_id', 'restorePointInTime']
 
 with ParametersContext(command='sql db') as c:
-    c.register_alias('database_name', ('--name', '-n'))
+    c.argument('database_name', options_list=('--name', '-n'), help='Name of the Azure SQL Database.')
+    c.argument('server_name', arg_type=server_param_type)
+    c.register_alias('elastic_pool_name', ('--elastic-pool',))
+    c.register_alias('requested_service_objective_name', ('--service-objective',))
+    c.register_alias('requested_service_objective_id', ('--service-objective-id',))
+
+def configure_db_create_params(c):
+    from azure.mgmt.sql.models.database import Database
+    c.expand('parameters', Database)
+
+    # We have a wrapper function that determines server location so user doesn't need to specify it as param.
+    # Also we have specific commands to special create modes which have their own required params, so
+    # database properties related to create mode can be ignored
+    ignores = ['location'] + sql_db_special_create_mode_params
+    for p in ignores:
+        c.ignore(p)
+
+with ParametersContext(command='sql db create') as c:
+    configure_db_create_params(c)
+
+# The following params are ignored when creating a copy or secondary because their
+# values are determined by the source db
+sql_db_copy_ignored_params = ['collation', 'edition', 'max_size_bytes']
```

Comment:
I don't think so. Python doesn't have forward references (i.e. call a function before it's defined) so I think the order in a file is just up to your choice based on the circumstances.